app/javascript/src/lib/codeActionProviders/transformParameterObjectsIntoPositionalParameters.js

```diff
@@ -0,0 +1,110 @@
+import { completionsMap, settings } from "../../stores/editor"
+import { getClosingBracket, getPhraseFromPosition, splitArgumentsString } from "../../utils/parse"
+import { get } from "svelte/store"
+import { parseParameterObjectContent } from "../../utils/compiler/parameterObjects"
+import { getIndentForLine } from "../../utils/codemirror/indent"
+
+/**
+ * @type {import("../codeActions").CodeActionProvider}
+ */
+export function transformParameterObjectsIntoPositionalParameters({ view, state }) {
+  const actions = []
+
+  const { from: cursorFrom, to: cursorTo } = state.selection.main
+  if (cursorFrom !== cursorTo) return
+
+  const line = state.doc.lineAt(cursorFrom)
+  if (!line.text) return null
+
+  const phrase = getPhraseFromPosition(line, cursorFrom)
+  const completion = get(completionsMap).find(({ type, label }) => type === "function" && label === phrase.text)
+  if (!completion) return
+
+  const fileWideActionNameStart = line.from + phrase.start
+  const fileWideActionNameEnd = line.from + phrase.end + 1
+
+  const actionCallCont = state.doc.sliceString(fileWideActionNameEnd, fileWideActionNameEnd + 100)
+  const openParenIndex = actionCallCont.match(/(?<=^\s*)\(/)?.index ?? -1
+  if (openParenIndex < 0) return
+
+  const closingParenIndex = getClosingBracket(" " + actionCallCont, "(", ")", 0) // this function is cursed...
+  if (closingParenIndex < 0) return
+
+  const fileWideParenFrom = fileWideActionNameEnd + openParenIndex
+  const fileWideParenTo = fileWideActionNameEnd + closingParenIndex
+
+  // empty positional parameters
+  if (fileWideParenTo - fileWideParenFrom <= "()".length) return
+
+  const openParameterObjectBracketIndex = actionCallCont.slice(openParenIndex + 1).match(/(?<=^\s*)\{/)?.index ?? -1
+  if (openParameterObjectBracketIndex >= 0) {
+    // parameter object → positional parameters
+
+    const closingParameterObjectBracketIndex = getClosingBracket(actionCallCont, "{", "}", 0)
+    if (closingParameterObjectBracketIndex < 0) return
+
+    const fileWideParameterObjectFrom = (fileWideParenFrom + 1) + openParameterObjectBracketIndex
+    const fileWideParameterObjectTo = (fileWideParenFrom + 1) + closingParameterObjectBracketIndex
+
+    actions.push({
+      label: "Transform object into positional parameters",
+      position: fileWideActionNameStart,
+      run() {
+        const parameterObject = state.doc.sliceString(fileWideParameterObjectFrom + 1, fileWideParameterObjectTo - 1)
```

Comment:
It seems a bit odd that this also does + 1, which is also added to the variable a couple lines before.  Is that correct?

Reply:
Yes, the +1 and -1 here are to get the stuff inside the curly braces (the "inner content" of the parameter objects).